File: python/7 kyu/55.py
# https://www.codewars.com/kata/5a04133e32b8b998dc000089/train/python

# Passed

def solve(arr, current_array=[]):
    if arr == current_array:
        return current_array
    
    array_to_return = list()
    for idx, num in enumerate(arr):
        if idx == len(arr) - 1:
            # last element
            array_to_return.append(num)
            return solve(array_to_return, current_array=arr)
        
        if num > arr[idx + 1]:
            array_to_return.append(num)

output = solve([92,52,93,31,89,87,77,105])
print(output)

File: nodejs/7 kyu/55.js
// https://www.codewars.com/kata/5e4217e476126b000170489b/train/javascript

// Passed

function polydivisible(x) {
  const digits = String(x).split("");
  const numbers = digits.map((value, idx) => digits.slice(0, idx + 1).join(""));
  const isPolyDivisible = numbers.every(
    (number, idx) => number % (idx + 1) === 0
  );
  return isPolyDivisible;
}

const output = polydivisible(123);
console.log(output);
